olive/hardware/accelerator.py

```diff
@@ -33,16 +33,19 @@ class AcceleratorSpec:
     """Accelerator specification is the concept of a hardware device that be used to optimize or evaluate a model."""
 
     accelerator_type: Union[str, Device]
-    execution_provider: str
+    execution_provider: Optional[str] = None
     vender: str = None
     version: str = None
     memory: int = None
     num_cores: int = None
 
     def __str__(self) -> str:
-        # remove the suffix "ExecutionProvider", len("ExecutionProvider") = 17
-        ep = self.execution_provider[:-17] or self.execution_provider
-        return f"{str(self.accelerator_type).lower()}-{ep.lower()}"
+        if self.execution_provider:
+            # remove the suffix "ExecutionProvider", len("ExecutionProvider") = 17
+            ep = self.execution_provider[:-17]
+            return f"{str(self.accelerator_type).lower()}-{ep.lower()}"
```

Comment:
Could we append `ep`? Make the output folder be like: cpu_cpuep_xx, gpu_cudaep_xxx

Emma also mentioned this before, when she tried to find the output model, she got confused when saw cpu_cpu_xx, gpu_cpu_xxx.

Appending `ep` may be better?

Reply:
I tried to add "ep" suffix but it seems all the current cache folder structure is impacted. I will stick to existing behavior to avoid feature broken.